src/modm/platform/uart/stm32/uart_hal_impl.hpp.in

```diff
@@ -63,10 +63,14 @@ void
 %% if over8
 	constexpr uint32_t scalar = (baudrate * 16l > SystemClock::{{ uart_name ~ id }}) ? 8 : 16;
 	constexpr uint32_t max = ((scalar == 16) ? (1ul << 16) : (1ul << 15)) - 1ul;
+	constexpr auto result = Prescaler::from_range(SystemClock::{{ uart_name ~ id }}, baudrate, 1, max);
+%% elif uart_name == "Lpuart"
+	constexpr uint32_t max = (1ul << 19) - 1ul;
+	constexpr auto result = Prescaler::from_range(SystemClock::{{ uart_name ~ id }} * 256, baudrate / 6, 1, max);
```

Comment:
I have no explanation fot this magic `/6` factor right now...

Reply:
Yeah, I was suspicious from the beginning, since the 19-bit - 16-bit = 3-bit, but the scalar is 256 = 8bit, so where do the other 5-bits go? But if this works, meh.

Reply:
I guess the `/6` is not the correct solution, it is just a coincidence. The optimal prescaler for 115200 baud is 0x2b672 (177778), ST's table in the manual says 0x2b671 (177777). The code in this PR with `/6` gives 171882. This looks pretty suspicious, still somehow close but not optimal.

Reply:
This is an int overflow problem, we are calculating prescalers in 32 bit arithmetic.
```
>>> (80e6*256)/(2**31 - 1)
9.536743168503392
```

Reply:
Oh crap, we should replace the `modm::frequency_t` and `uint32_t` with `uint64_t` in the `modm::Prescaler` class:
https://github.com/modm-io/modm/blob/develop/src/modm/math/algorithm/prescaler.hpp

Reply:
We have to be careful, int overflow is evil. This compiles:
```
static constexpr uint32_t clock = 80'000'000;
static constexpr uint64_t clock2 = 80'000'000;

static_assert((clock * 256) != (clock2 * 256));
```

The system clock structs need an update as well.

Reply:
What do you think about my solution in adbe4da?

```diff
-class
-Prescaler
+template<typename T>
+class
+GenericPrescaler
{
// ...
// sed s/modm::frequency_t/T/
// ...
};

+using Prescaler = GenericPrescaler<modm::frequency_t>;
```

Reply:
It's good, it leaves the option open to use it at runtime with 32-bit, however I would just make it `GenericPrescaler<uint64_t>;` by default, since we only use it at compile time.